Optimize channel_stats_op

xiaomengy opened the pull request:
Summary: Optimize channel_stats_op and add NHWC impl

Differential Revision: D13775515

BIT-silence commented:
About 2x faster than before on GPU

houseroad commented on caffe2/python/operator_test/channel_stats_op_test.py:
```diff
@@ -1,42 +1,85 @@
-                .astype(np.float32) - 0.5
-        self.assertReferenceChecks(gc, op, [X], referenceChannelStatsTest)
+        X = np.random.randn(N, C, D, H, W).astype(np.float32)
+        if order == "NCHW":
```
Should this be NHWC?

houseroad reviewed (commented):
Otherwise, looks good.

xiaomengy committed:
Optimize channel_stats_op (#16243)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/16243

Optimize channel_stats_op and add NHWC impl

Reviewed By: takatosp1

Differential Revision: D13775515

fbshipit-source-id: 91298117056ed9d47bd2f842d8c17e694dcf6902

houseroad reviewed (approved):
Looks good.
